flink-runtime/src/main/java/org/apache/flink/runtime/rpc/akka/AkkaRpcService.java

```diff
@@ -315,33 +315,48 @@ public void stopServer(RpcServer selfGateway) {
 			}
 
 			if (rpcEndpoint != null) {
-				akkaClient.getActorRef().tell(PoisonPill.getInstance(), ActorRef.noSender());
+				terminateAkkaRpcActor(akkaClient.getActorRef(), rpcEndpoint);
 			} else {
 				LOG.debug("RPC endpoint {} already stopped or from different RPC service", selfGateway.getAddress());
 			}
 		}
 	}
 
+	private CompletableFuture<Void> terminateAkkaRpcActor(ActorRef akkaRpcActorRef, RpcEndpoint rpcEndpoint) {
+		akkaRpcActorRef.tell(ControlMessages.TERMINATE, ActorRef.noSender());
+
+		return rpcEndpoint.getTerminationFuture();
+	}
+
 	@Override
 	public CompletableFuture<Void> stopService() {
+		LOG.info("Stopping Akka RPC service.");
```

Comment:
Depending on what you want to see from this message, it could be more precise to move the logging under the if `if` statement that starts in 338.

Reply:
true, will change it.